### PR TITLE
Added test_image_bounder

### PR DESCRIPTION
**Pull request by Inffzy**

### CODE REVIEW

**Comment by Inffzy** [comment]
Code runs with no errors by running python -m nose unit_tests from tests. 
Problems occurred from running python -m nose tests/unit_tests from SUAS-Competition are yet to be fixed.

File: SyntheticDataset/SyntheticDataset/image_operations/ImageBounder.py [hunk 1]
```diff
@@ -5,6 +5,26 @@ class ImageBounder(object):
...
+                if  image[x,y] == color:
+                    list_of_x.append(x)
+                    list_of_y.append(y)
+
+        leftX = min(list_of_x)
+        rightX = max(list_of_x)
+        upY = min(list_of_y)
+        lowY = max(list_of_y)
+
+        return Rectangle(leftX-1, upY-1, rightX+1, lowY+1)
```
Comment (DarkmatterVale) [inline SyntheticDataset/SyntheticDataset/image_operations/ImageBounder.py]: This is not correct. If you look at the rectangle documentation, you need to pass the top left point, then the width and height of the rectangle (not the bottom right point).

File: SyntheticDataset/SyntheticDataset/image_operations/ImageBounder.py [hunk 2]
```diff
@@ -5,6 +5,26 @@ class ImageBounder(object):
...
+
+        """
```
Comment (DarkmatterVale) [inline SyntheticDataset/SyntheticDataset/image_operations/ImageBounder.py]: If you are commenting out code, simply remove it. It is bad practice to just comment out code when you replace it.

File: tests/unit_tests/test_image_bounder.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+from SyntheticDataset.image_operations import *
+import unittest
+from PIL import Image
+
+class ImageBounderTestCase(unittest.TestCase):
+
+
+    def test_getBoundsOfColor(self):
+        self.test_img1 = Image.open("tests/images/test_image2.png")
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_image_bounder.py]: Please rename the images to something more appropriate. The names should relate to the test case with which they are associated to.

File: tests/unit_tests/test_image_bounder.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
+from SyntheticDataset.image_operations import *
+import unittest
+from PIL import Image
+
+class ImageBounderTestCase(unittest.TestCase):
+
+
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_image_bounder.py]: Remove the second line of spacing here.

File: tests/unit_tests/test_image_bounder.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    def test_getBoundsOfColor(self):
+        self.test_img1 = Image.open("tests/images/test_image2.png")
+        self.test_image1 = self.test_img1.load()
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_image_bounder.py]: If you are going to assign the loaded image to the same variable that contained the original image, condense the two calls into one line.

File: tests/unit_tests/test_image_bounder.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        self.test_color1 = (255,255,255,255)
+        self.test_img2 = self.test_img1.copy()
+        self.test_image2 = self.test_img2.load()
+
+        print (ImageBounder.getBoundsOfColor(self.test_img1, self.test_image1, self.test_color1))
+
+        #self.assertEqual(ImageBounder.getBoundsOfColor(self.test_img1, self.test_image1, self.test_color1), Rectangle(-1,-1,322,242))
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_image_bounder.py]: If you are not using this code, please remove it.

File: tests/unit_tests/test_image_bounder.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        self.assertEqual(ImageBounder.getBoundsOfColor(self.test_img3, self.test_image3, self.test_color3).getHeight(), 416)
+
+        self.test_resultant3 = self.test_img3.crop((111, -1, 208, 416))
+        self.test_resultant3.show()
+
+        self.test_color4 = (83,186,131)
+
+        print (ImageBounder.getBoundsOfColor(self.test_img4, self.test_image4, self.test_color4))
+
+        #self.assertEqual(ImageBounder.getBoundsOfColor(self.test_img4, self.test_image2, self.test_color4), Rectangle(319,-1,322,242))
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_image_bounder.py]: If you are not using this code, please remove it.

**Review by DarkmatterVale — CHANGES_REQUESTED** [review]
Getting much better! Everything looks good except for whitespace, comments, and image names.

File: tests/unit_tests/test_image_bounder.py (hunk 5 above)
Comment (Inffzy) [reply tests/unit_tests/test_image_bounder.py]: i do need both img and image here

Comment (DarkmatterVale) [reply tests/unit_tests/test_image_bounder.py]: Ah yes you are correct I apologize...For some reason my brain thought those two images were both the same name...

**Review by DarkmatterVale — APPROVED** [review]

**Pull request merged**